### DISCUSSION TITLE
Customize Sources - add an event / settings to change sources in production environment

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#11536] [Ideas] [answered]
In my opinion moving all settings for custom sources into the project config was not a good thing. It was an awesome step in Craft 3 to let each user have their own setting and I have no idea why this was reverted. 

I already spend a few hours to kinda revert your change.
The `CustomizeSourcesModal` as well as your save/update actions can be enhanced really easily to be able to store these information in the DB/localStorage again rather than project config but the 
```twig
{% if currentUser.admin and craft.app.config.general.allowAdminChanges %}
```
condition in your `elementindexes.twig` is a little bit more difficult.
Would you please add an event or a config value to be able to display the `edit sources` button again so we can change the table columns on our requirements?

Actually I would really love to revert that like it was in Craft 3. Our clients wish that too and it's my current task to do this before we use Craft 4 more.

Just an explanation why that's kinda a dealbreaker:
my lists as an admin are nearly always entirely different from the one our clients need. They rather need useful information about the content while I mostly need relevant information for the code / filters / logic I build or check.
In 99% of use cases I need to see a bunch of columns that are relevant for my code and the requirement to always revert this after I do something so our clients won't have a messed up view is incredibly tedious. There is no reason for them to have their tables changed during every single update and I don't want these information in the project config at all. All that's left to make this working is the hard check in your twig template and that's little bit more difficult to extend /overwrite in a clean way.

**Comment by brandonkelly** [comment, marked as answer]
> It was an awesome step in Craft 3 to let each user have their own setting and I have no idea why this was reverted.

That was never the case. Non-admin users could be granted permission to change the _global_ source settings, but it would affect all users, not just their own account.

We are considering making it possible for users to customize it to their own needs, though – see #11355.

> Would you please add an event or a config value to be able to display the `edit sources` button again so we can change the table columns on our requirements?

Not exactly what you’re asking for, but it’s possible to modify the source settings at runtime via the `craft\base\Element::EVENT_REGISTER_SOURCES` event.
